value/complex.go

```diff
@@ -191,3 +191,40 @@ func (c Complex) div(ctx Context, d Complex) Complex {
 	i = ctx.EvalBinary(i, "/", denom)
 	return NewComplex(r, i)
 }
+
+// floor returns the complex floor, as defined by
+// McDonnell, E. E. "Complex Floor, APL Congress 73." (1973).
+// https://www.jsoftware.com/papers/eem/complexfloor.htm
+//
+//	op floor z =
+//		r = real z
+//		i = imag z
+//		b = (floor r) j (floor i)
+//		x = r mod 1
+//		y = i mod 1
+//		1 > x + y : b
+//		x >= y    : b + 1
+//		b + 0j1
```

Comment:
Given that the algorithm is written right below this, and is rather long as you've expressed it, perhaps you don't need this. People truly interested can see the paper. How about
// floor returns the complex floor, as defined by
// McDonnell, E. E. "Complex Floor, APL Congress 73." (1973).
// https://www.jsoftware.com/papers/eem/complexfloor.htm
// The rough idea is to find the nearest corner of the unit square
// containing the point, rounding towards the origin.